File: result/metagpt/claude/humaneval/12.c
#include <stdio.h>
#include <string.h>

char* longest(char** strings, int size) {
    if (size == 0) {
        return NULL;
    }

    char* out = strings[0];
    for (int i = 1; i < size; i++) {
        if (strlen(strings[i]) > strlen(out)) {
            out = strings[i];
        }
    }
    return out;
}

File: result/metagpt/gpt4o/humaneval/12.rs
fn longest(strings: &[&str]) -> Option<&str> {
    if strings.is_empty() {
        return None;
    }

    let mut out = strings[0];
    for &string in &strings[1..] {
        if string.len() > out.len() {
            out = string;
        }
    }
    Some(out)
}

fn main() {
    let strings = vec!["short", "longer", "longest"];
    match longest(&strings) {
        Some(longest_str) => println!("Longest string: {}", longest_str),
        None => println!("No strings provided."),
    }
}